handler/src/main/java/io/netty/handler/timeout/WriteTimeoutHandler.java

```diff
@@ -111,9 +115,19 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
         ctx.write(msg, promise);
     }
 
+    @Override
+    public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
+        for (ScheduledFuture<?> timeout: timeouts) {
+            timeout.cancel(false);
+        }
+        timeouts.clear();
+    }
+
     private void scheduleTimeout(final ChannelHandlerContext ctx, final ChannelPromise future) {
         // Schedule a timeout.
-        final ScheduledFuture<?> sf = ctx.executor().schedule(new OneTimeTask() {
+        // Create an array to access timeout[0] in the inner class before assigning ScheduledFuture
+        final ScheduledFuture<?>[] timeout = new ScheduledFuture<?>[1];
```

Comment:
I guess the array is used @windie tries to access the future that is returned by `schedule(...)` in the `OneTimeTask` itself. Consider just using create a `OneTimeTask` subclass which will contain the `Future` in a field. Something like this:

```java
private class TimeoutTask extends OneTimeTask {
    Future<?> future;
    
    @Override
    public void run() {
        ...
        if (future != null) {
            timeouts.remove(future); 
        }
    }
}

....
TimeoutTask task = new TimeoutTask();

task.future = ctx.executor().schedule(task, timeoutNanos, TimeUnit.NANOSECONDS);
...
```